### PR TITLE
Add support for gt, gte, lt, lte operators for numeric, bool and date…

### PR DESCRIPTION
**Pull request by kchoy-sfdc**
… time data type in where clause

### CODE REVIEW

File: marshaller.go [hunk 1]
```diff
@@ -131,10 +152,23 @@ func constructLikeClause(v interface{}, fieldName string, exclude bool) string {
 
 func buildInClause(v interface{}, fieldName string) string {
 	var buff strings.Builder
-	items, ok := v.([]string)
-	if !ok {
+	var items []string
+	inSingleQuotes := false
```
Comment (atulkc) [inline marshaller.go]: May be rename the variable to `useSingleQuotes`? Just makes it more descriptive

File: marshaller.go [hunk 2]
```diff
@@ -162,18 +200,49 @@ func buildEqualsClause(v interface{}, fieldName string) string {
 	return constructEqualsClause(v, fieldName, equalsOperator)
 }
 
+func buildGreaterThanClause(v interface{}, fieldName string) string {
+	return constructEqualsClause(v, fieldName, greaterThanOperator)
```
Comment (atulkc) [inline marshaller.go]: Can we rename `constructEqualsClause` to `constructConditionalClause` or something that depicts that it is constructing more than just equals clause?

File: marshaller_test.go [hunk 3]
```diff
@@ -286,6 +288,123 @@ var _ = Describe("Marshaller", func() {
 			})
 		})
 
+		Context("when all clauses are signed integer data types", func() {
```
Comment (atulkc) [inline marshaller_test.go]: Can we also add test for mixed types? Query criteria that has some string, some integer, some time fields.

File: marshaller_test.go [hunk 4]
```diff
@@ -286,6 +288,123 @@ var _ = Describe("Marshaller", func() {
...
+				expectedClause = "Num_of_CPU_Cores__c > 16 AND Physical_CPU_Count__c < 4 AND Number_Of_Successive_Puppet_Run_Failures__c >= 0 AND Num_Of_Coolan_Log_Files__c <= 1024"
+			})
+
+			It("returns properly formed clause", func() {
+				clause, err = MarshalWhereClause(criteria)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(clause).To(Equal(expectedClause))
+			})
+		})
+
```
Comment (atulkc) [inline marshaller_test.go]: Also, lets add one test to `Describe("Marshal")` that constructs complete SOQL with query criteria of having mixed types (string, int, time...etc).

File: marshaller.go (hunk 1 above)
Comment (kchoy-sfdc) [reply marshaller.go]: Sure

File: marshaller.go (hunk 2 above)
Comment (kchoy-sfdc) [reply marshaller.go]: How does `constructComparisonClause` sound?

File: marshaller_test.go (hunk 3 above)
Comment (kchoy-sfdc) [reply marshaller_test.go]: Will do

File: marshaller_test.go (hunk 4 above)
Comment (kchoy-sfdc) [reply marshaller_test.go]: Will do

**Pull request merged**